Audacia.Quest/Audacia.Quest.Core/Components/BaseComponent.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Audacia.Quest.Core.Asset;
+
+namespace Audacia.Quest.Core.Components
+{
+    public abstract class BaseComponent : IComponent
+    {
+        public Guid Id { get; set; } = Guid.NewGuid();
+
+        public Transform Transform { get; set; } = new Transform();
+        public IComponent Parent { get; set; }
+        public List<ComponentMap> Components { get; set; } = new List<ComponentMap>();
+        public Sprite Sprite { get; set; }
+
+        private bool _enabled = true;
+
+        public bool Enabled
+        {
+            get
+            {
+                return _enabled;
+            }
+            set
+            {
+                var oldValue = _enabled;
+                _enabled = value;
+
+                if (value && oldValue != value)
+                {
+                    Init();
+                }
+            }
+        }
+
+        public IComponent WithRenderer(Sprite sprite)
+        {
+            Sprite = sprite;
+            return this;
+        }
+
+        public IComponent WithComponent(IComponent component, bool enabled = true)
+        {
+            component.Enabled = enabled;
+            AddComponent(component);
+            return this;
+        }
+
+        public void AddComponent(IComponent component)
+        {
+            if (component == null)
+            {
+                return;
+            }
+
+            component.Parent = this;
+            var type = component.GetType();
+            Components.Add(new ComponentMap
+            {
+                Name = type.Name,
+                Component = component
+            });
+        }
+
+        public TComponent? GetComponent<TComponent>()
+            where TComponent : IComponent
+        {
+            var name = typeof(TComponent).Name;
+
+            var component = Components.FirstOrDefault(x => x.Name == name);
+            if (component != null)
+            {
+                return (TComponent)component.Component;
+            }
+
+            return default;
+        }
+
+        public List<TComponent> GetComponents<TComponent>()
```

Comment:
This might be a personal preference, but I'd recommend returning an interface instead of a type for Lists and Arrays.
So maybe `IList<T>`, `IEnumerable<T>`, `ICollection<T>` etc, If it was explicitly an array then `IReadOnlyCollection<T>`.